File: mobileBE/src/main/java/com/uth/mobileBE/dto/response/BookDetailInfoDto.java
package com.uth.mobileBE.dto.response;

import com.uth.mobileBE.models.enums.StatusLoanDetail;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BookDetailInfoDto {
    private Long copyId;         // RẤT QUAN TRỌNG: Dùng để gọi API Sửa/Xóa từng cuốn
    private String title;
    private String author;
    private String category;
    private LocalDateTime dueDate;
    private LocalDateTime returnDate;
    private StatusLoanDetail status; // Hoặc String nếu bạn đã đổi sang String
}

File: app/src/main/java/com/example/quanlythuvien/data/model/response/BookDetailInfoDto.kt
package com.example.quanlythuvien.data.model.response

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime


data class BookDetailInfoDto(
    @SerializedName("copyId")
    val copyId: Long,

    @SerializedName("title")
    val title: String?, // Nên để nullable đề phòng DB bị null

    @SerializedName("author")
    val author: String?,

    @SerializedName("category")
    val category: String?,

    @SerializedName("dueDate")
    val dueDate: String?,

    @SerializedName("returnDate")
    val returnDate: String?,

    @SerializedName("status")
    val status: String?
)